MainWindow.xaml.cs

```diff
@@ -569,7 +621,7 @@ private void EnsureDiscInformation()
         private void GetOutputNames()
         {
             var driveTuple = cmb_DriveLetter.SelectedItem as Tuple<char, string, bool>;
-            var discTuple = cmb_DiscType.SelectedItem as Tuple<string, KnownSystem?, DiscType?>;
+            var discTuple = cmb_SystemType.SelectedItem as Tuple<string, KnownSystem?, DiscType?>;
```

Comment:
This one needs to be `Tuple<string, KnownSystem?>`

Reply:
Actually we don't need to fetch anything again here since we already have `selectedSystem` and `selectedDiscType` from above, I'll enhance accordingly.

Reply:
This would still be incorrect as `cmb_SystemType.SelectedItem` is a `Tuple<string, KnownSystem?>` and this would return null always.

Reply:
yes, I was actually talking about something similar in `EnsureDiscInformation()` method, here we need to fetch them.